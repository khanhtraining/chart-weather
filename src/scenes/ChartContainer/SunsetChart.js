import React, { useEffect, useRef } from 'react'
import { select } from 'd3'
import * as d3 from 'd3'

const SunsetChart = (props) => {
    const ref = useRef()
    const { width, height, data } = props
    const x = d3.scalePoint()
        .domain(((data).map(function (d) { return d.hour })))
        .range([0, width])
    const y = d3.scaleLinear()
        .domain([0, 3])
        .range([height, 0])

    useEffect(() => {
        const g = select(ref.current)
        const lineSunPath = d3.area()
            .x(function (d) { return x(d.hour) })
            .y(function (d) { return y(d.sun) })
            .curve(d3.curveMonotoneX)
        g.append('path')
            .data([(data)])
            .attr('class', 'line-sun')
            .style("stroke", 'orange')
            .style("stroke-width", '2px')
            .style("fill", 'none')
            .attr('d', lineSunPath)
    }, [width, height, data, x, y])

    return (
        <g className='sunset-chart' ref={ref} />
    )
}

export default SunsetChart
