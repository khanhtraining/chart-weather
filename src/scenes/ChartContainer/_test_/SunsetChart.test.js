import React from 'react'
import renderer from 'react-test-renderer'
import SunsetChart from '../SunsetChart'

describe('snapshot testing', () => {
    test('snapshot for sunset chart component', () => {
        const renderedComponent = renderer.create(<SunsetChart/>).toJSON()
        expect(renderedComponent).toMatchSnapshot()
    })
})
