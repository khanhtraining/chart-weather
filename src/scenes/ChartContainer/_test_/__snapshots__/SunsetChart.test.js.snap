// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for sunset chart component 1`] = `
<g
  className="sunset-chart"
/>
`;
